### PR TITLE
Add Internal Helper for Deserializing Subclasses

### LINKED ISSUE #4346: [FEATURE] New `TO.de_json_subclasses` method for parent classes
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

We can save some code duplication by extracting the `de_json` logic of classes like `PaidMedia` or `ChatMember`. Ideally this should go into `TelegramObject` so we can easily call it.

### Describe the solution you'd like

https://github.com/python-telegram-bot/python-telegram-bot/pull/4334#discussion_r1665902751

### Describe alternatives you've considered

keep doing what we do right now

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Closes #4346

TBF, this is more clumsy than I had hoped, but if TG continues with the pattern of subclasses it should still save some duplication.

It's a +0 from my side. If you find the logic hard to follow and think that it makes debugging harder, I'm okay with rejecting.

### CODE REVIEW

**Review by harshil21 — COMMENTED** [review]
oh.. looks like it became much more complicated with very little to no benefit.. (LoC is also greater). I also propose to scrap.

**Comment by Bibo-Joshi** [comment]
Fair enough. I'll close and just open a new PR to add the test to menubutton :)
